### PR TITLE
puppet-lint: Enable whitespace checker

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi people,
please comment on the diff. I would like to add https://github.com/kuleuven/puppet-lint-manifest_whitespace-check to our testmatrix. The whole diff is caused by the autofix. I'm currently tracking down one bug: https://github.com/kuleuven/puppet-lint-manifest_whitespace-check/issues/3

**Comment by bastelfreak** [comment]
upstream bug: https://github.com/kuleuven/puppet-lint-manifest_whitespace-check/issues/4

File: examples/plugins/ceph.pp [hunk 1]
```diff
@@ -1,7 +1,5 @@
 include collectd
 
 class { 'collectd::plugin::ceph':
-  osds          => [ 'osd.0', 'osd.1', 'osd.2'],
+  osds          => ['osd.0', 'osd.1', 'osd.2'],
```
Comment (alexjfisher) [inline examples/plugins/ceph.pp:4]: I thought the linter was also going to pick up the crazy about of extra whitespace here?

File: manifests/plugin/apache.pp [hunk 2]
```diff
@@ -2,11 +2,10 @@
 class collectd::plugin::apache (
   Enum['present', 'absent'] $ensure        = 'present',
   Boolean $manage_package                  = $collectd::manage_package,
-  Hash $instances                          = { 'localhost' => { 'url' => 'http://localhost/mod_status?auto' } },
+  Hash $instances                          = { 'localhost' => { 'url' => 'http://localhost/mod_status?auto' }},
```
Comment (alexjfisher) [inline manifests/plugin/apache.pp:5]: Why is this space removed, but not the others?

File: manifests/params.pp [hunk 3]
```diff
@@ -33,12 +32,12 @@
 
   case $facts['kernel'] {
     'OpenBSD': { $has_wordexp = false }
-    default:   { $has_wordexp = true }
+    default: { $has_wordexp = true }
```
Comment (alexjfisher) [inline manifests/params.pp:35]: It made sense to align the cases before though?

File: examples/plugins/ceph.pp (hunk 1 above)
Comment (jovandeginste) [reply examples/plugins/ceph.pp:4]: I thought there was a different linter for this. I may be mistaken... Maybe it only verifies whether they are vertically aligned (which is always true when there is only one element)

File: manifests/params.pp (hunk 3 above)
Comment (jovandeginste) [reply manifests/params.pp:35]: Agreed

File: manifests/plugin/apache.pp (hunk 2 above)
Comment (jovandeginste) [reply manifests/plugin/apache.pp:5]: As a side note, I had a similar case where I had a default value for a Hash (but more data and thus multiline), which crashed [class_parameter-check](https://rubygems.org/gems/puppet-lint-class_parameter-check)

File: examples/plugins/ceph.pp (hunk 1 above)
Comment (bastelfreak) [reply examples/plugins/ceph.pp:4]: that should be fixed with this plugin: https://github.com/voxpupuli/puppet-collectd/pull/948

File: Gemfile [hunk 4]
```diff
@@ -11,7 +11,8 @@ def location_for(place, fake_version = nil)
 end
 
 group :test do
-  gem 'voxpupuli-test', '>= 1.0.0',  :require => false
+  gem 'voxpupuli-test', '>= 1.4.0',  :require => false
+  gem 'puppet-lint-manifest_whitespace-check', :require => false
```
Comment (ekohl) [inline Gemfile:15]: I'd prefer to make this a part of vp-test 1.5.0 if it works well (which it appears to do)

Comment (bastelfreak) [reply Gemfile:15]: yes, me too. I just added it for testing / enable people to reproduce this.

**Comment by ekohl** [comment]
This was included in modulesync 3.0.0

**Pull request closed without merging**